kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -245,10 +246,40 @@ public List<DomainMetadata> getDomainMetadatas() {
             dataPath.toString(), domainName, readSnapshot.getVersion());
       }
     }
-    domainMetadatas = Optional.of(finalDomainMetadatas);
+    domainMetadatasToCommit = Optional.of(finalDomainMetadatas);
     return finalDomainMetadatas;
   }
 
+  /**
+   * Returns the set of active domain metadata of the table, removed domain metadata are excluded.
+   */
+  private Optional<Set<DomainMetadata>> getActiveDomainMetadatas() {
```

Comment:
IMO this Transaction class is getting too big and it is violating the single responsibility principle.

For example, when this says "active" --> does that mean active at the time before the transaction has started? active after the transaction has completed? I know it means the latter -- but it is fair to interpret `Transaction:: getActiveDomainMetadatas` as the former.

Can you think of a way to separate this out to another class? Perhaps most of the post-transaction stuff?

Reply:
We can punt that to another PR -- but at the very least here
(1) rename to `getPostCommitDomainMetadatas`
(2) move this private helper method to the bottom of the class where the private members are

Reply:
Renamed and moved the helper